core/src/main/java/org/springframework/security/authentication/AbstractAuthenticationToken.java

```diff
@@ -222,4 +222,16 @@ public String toString() {
 
 		return sb.toString();
 	}
+
+	protected static Integer extractKeyHash(String key) {
+		Object value = nullSafeValue(key);
+		return value.hashCode();
+	}
+
+	protected static Object nullSafeValue(Object value){
```

Comment:
While it provides re-use, this method does not make sense in `AbstractAuthenticationToken`. Instead, we should move this to a private method within each subclass.